### DISCUSSION TITLE
New status conditions for unsupported (non-core) features

### DISCUSSION THREAD

**Discussion by mlavacca** [kubernetes-sigs/gateway-api#1826] [Q&A] [answered]
Since the API has a bunch of non-core features (either extended or experimental), and [GEP-1709](https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-1709.md) plans to assert the various types of conformance through a set of profiles, it could be helpful to introduce a new status condition related to the non-core unsupported features, to standardize the way the user gets feedback about not-supported Gateway API features.

For example, if a user creates an HTTPRoute with a [`ResponseHeaderModifier` filter](https://github.com/kubernetes-sigs/gateway-api/blob/be9c607814ff73c375074b796016e66e2268618e/apis/v1beta1/httproute_types.go#L602-L609) set, but the implementation does not support such a feature, it could be useful to have a standard and shared way to state that the HTTPRoute is not valid because the feature is experimental and not supported by the implementation.

**Comment by shaneutt** [comment, marked as answer]
I believe this need is already captured by https://github.com/kubernetes-sigs/gateway-api/issues/1804 (which as it stands we're pushing to resolve _before_ GA). Can you look that issue over and let me know if you feel it doesn't fully cover what you want, or if there are ways we can improve the wording to better include your needs?

**Reply by mlavacca** [reply]
That issue is precisely what I was looking for; thanks for pointing me!
